tutorials/monitoring-iot-data-grafana/index.md

```diff
@@ -0,0 +1,403 @@
+
+    open "https://console.cloud.google.com/billing/linkedaccount?project=$PROJECT_ID"
+
+### Enable services
+
+Enable Pub/Sub and IoT Core services in your project: 
+
+    gcloud services enable cloudiot.googleapis.com pubsub.googleapis.com
+
+### Grant publishing role for Pub/Sub
+
+You need to permit IoT Core to publish to the Pub/Sub service. Because IoT Core is responsible for your devices, and they don’t need to subscribe to any topics,
+you give them just the publishing role:
+
+    gcloud projects add-iam-policy-binding $PROJECT_ID \
+        --member=serviceAccount:cloud-iot@system.gserviceaccount.com \
+        --role=roles/pubsub.publisher
+
+### Choose a region
+
+Choose a region as described [here](https://cloud.google.com/compute/docs/regions-zones/).
+
+This tutorial uses `us-central1`, but you can pick one that better suits you.
+
+### Set up a Pub/Sub topic, subscription, and registry
+
+On MQTT, Pub/Sub works like an Instagram or Twitter hashtag, where you can publish a post using a hashtag and whoever is following (or subscribed to)
+that hashtag sees your post. MQTT works in a similar way, but the hashtag is the topic, the photo is the message, and the people following that topic is the 
+subscription.
+
+A registry is like a bucket for your IoT devices. It allows you to group devices and set properties that they all share, such as connection protocol, data 
+storage location, and Pub/Sub topics.
+
+The following commands create a topic, subscription, and registry:
+
+    gcloud pubsub topics create $TOPIC_ID
+
+    gcloud pubsub subscriptions create --topic $TOPIC_ID $SUBSCRIPTION
+
+    gcloud iot registries create $REGISTRY \
+        --region=$REGION \
+        --event-notification-config=topic=temperature-topic \
+        --enable-mqtt-config --enable-http-config
+
+You can check your [registries](https://console.cloud.google.com/iot/registries) and, if everything goes well, you should see your registry with your topic and 
+subscription there.
+
+![registries](https://storage.googleapis.com/gcp-community/tutorials/monitoring-iot-data-grafana/img2.png)
+
+## Set up the ESP32 device
+
+This tutorial uses the Espressif micro-controller ESP32 for its WiFi and built-in temperature sensor. 
+
+### Generate a key pair for device authentication
+
+You need to generate an elliptic curve (EC) ES256 private/public key pair for your device authentication. Make sure to store them in a safe place.
+
+    openssl ecparam -genkey -name prime256v1 -noout -out ec_private.pem
+
+    openssl ec -in ec_private.pem -pubout -out ec_public.pem
+    
+### Register the device with IoT Core
+
+    gcloud iot devices create $DEVICE_ID \
+        --region=$REGION \
+        --registry=$REGISTRY \
+        --public-key="path=./ec_public.pem,type=es256"
+
+### Install Arduino libraries
+
+Install the “Google Cloud IoT Core JWT” and lwmMQTT from Joel Gaehwiler libraries on your Arduino IDE. These libraries are responsible for connecting, 
```

Comment:
I think that we should give folks a bit more information about how to get these libraries.

Link here?

https://github.com/256dpi/arduino-mqtt

Reply:
These are the links for [Google Cloud IoT Core](https://github.com/GoogleCloudPlatform/google-cloud-iot-arduino) and this is the [lwMQTT lib](https://github.com/256dpi/arduino-mqtt).

We can also include this images:

<img width="827" alt="Screen Shot 2020-10-29 at 01 09 33" src="https://user-images.githubusercontent.com/38125236/97524664-fcf11400-1983-11eb-8a7f-0774b0b3a612.png">

<img width="805" alt="Screen Shot 2020-10-29 at 01 09 06" src="https://user-images.githubusercontent.com/38125236/97524672-01b5c800-1984-11eb-9899-403b28149ab4.png">

<img width="803" alt="Screen Shot 2020-10-29 at 01 10 58" src="https://user-images.githubusercontent.com/38125236/97524676-04182200-1984-11eb-8f33-139635e4cb0f.png">

Reply:
Thanks. I added the links.